File: 1408.数组中的字符串匹配.cpp
/*
 * @lc app=leetcode.cn id=1408 lang=cpp
 *
 * [1408] 数组中的字符串匹配
 */
#include <string>
#include <vector>
using namespace std;
// @lc code=start
class Solution {
public:
  /**
   * @brief
   * 就这？
   * 67/67 cases passed (4 ms)
   * Your runtime beats 83.28 % of cpp submissions
   * Your memory usage beats 85.87 % of cpp submissions (8 MB)
   * @param words
   * @return vector<string>
   */
  vector<string> stringMatching(vector<string> &words) {
    auto ans = vector<string>();
    for (int i = 0; i < words.size(); i++) {
      for (int j = 0; j < words.size(); j++) {
        if (i == j)
          continue;
        if (words[i].length() > words[j].length())
          continue;
        if (words[j].find(words[i]) != string::npos) {
          ans.push_back(words[i]);
          break;
        }
      }
    }
    return ans;
  }
};
// @lc code=end

File: 1408.数组中的字符串匹配.java
/*
 * @lc app=leetcode.cn id=1408 lang=java
 *
 * [1408] 数组中的字符串匹配
 */
import java.util.ArrayList;
import java.util.List;

// @lc code=start
class Solution {

  /**
   * 67/67 cases passed (5 ms)
   * Your runtime beats 21.07 % of java submissions
   * Your memory usage beats 75.9 % of java submissions (40.1 MB)
   * @param words
   * @return
   */
  public List<String> stringMatching(String[] words) {
    var ans = new ArrayList<String>();
    for (var i = 0; i < words.length; i++) {
      for (var j = 0; j < words.length; j++) {
        if (i == j) {
          continue;
        }
        if (words[i].length() > words[j].length()) {
          continue;
        }
        if (words[j].contains(words[i])) {
          ans.add(words[i]);
          break;
        }
      }
    }
    return ans;
  }
}
// @lc code=end
